src/main/kotlin/it/pagopa/ecommerce/helpdesk/utils/ConfidentialMailUtils.kt

```diff
@@ -0,0 +1,54 @@
+package it.pagopa.ecommerce.helpdesk.utils
+
+import it.pagopa.ecommerce.commons.domain.Confidential
+import it.pagopa.ecommerce.commons.domain.Email
+import it.pagopa.ecommerce.commons.utils.ConfidentialDataManager
+import kotlinx.coroutines.reactor.mono
+import org.slf4j.LoggerFactory
+import reactor.core.publisher.Mono
+
+/**
+ * Class used for handle mail encryption end decryption.
+ *
+ * @param emailConfidentialDataManager class for execute pdv call
+ * @param emailCachedMap Map containing the opaque mail token as the key and the clear mail as the
+ *   value.
+ */
+class ConfidentialMailUtils(
+    private val emailConfidentialDataManager: ConfidentialDataManager,
+    private val emailCachedMap: MutableMap<String, Email> = mutableMapOf()
+) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    /**
+     * The method search clear mail into cache if not found make a call to pdv to decrypt the email
+     * then cache it
+     *
+     * @param encrypted email encrypted
+     * @return Mono<Email> return mono with clear email value object
+     */
+    fun toEmail(encrypted: Confidential<Email>): Mono<Email> {
```

Comment:
Add exception handler mapping for exception that can be raised by `ConfidentialDataManager.decrypt` and `ConfidentialDataManager.encrypt` and map to proper error response